### PR TITLE
Add support for libraries testing with NativeAOT

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
This adds support for running libraries tests with NativeAOT. It reuses the single file xunit runner since we cannot `LoadFrom` the tests on NativeAOT. The strategy is the same as for single file testing: do a publish. To do the NativeAOT publish, we include the Microsoft.DotNet.ILCompiler.targets.

Not attempting to add this to the CI since the tests are currently hitting #63778.

### CODE REVIEW

File: src/libraries/externals.csproj [hunk 1]
```diff
@@ -26,7 +26,8 @@
 
   <!-- Setup the testing shared framework host -->
   <Target Name="SetupTestingHost"
-          AfterTargets="AfterResolveReferences">
+          AfterTargets="AfterResolveReferences"
+          Condition="'$(TestNativeAot)' != 'true'">
```
Comment (MichalStrehovsky) [inline src/libraries/externals.csproj:30]: This and the below targets probably wouldn't harm, but also it's possible to build a usable NativeAOT subset without building the host or corerun and having these fail to copy the useless files in that state is an annoyance.

File: eng/testing/tests.singlefile.targets [hunk 2]
```diff
@@ -12,14 +12,37 @@
...
 
+  <PropertyGroup Condition="'$(TestNativeAot)' == 'true'">
+    <IlcToolsPath>$(CoreCLRILCompilerDir)</IlcToolsPath>
+    <IlcBuildTasksPath>$(CoreCLRILCompilerDir)netstandard/ILCompiler.Build.Tasks.dll</IlcBuildTasksPath>
+    <IlcSdkPath>$(CoreCLRAotSdkDir)</IlcSdkPath>
+    <IlcFrameworkPath>$(NetCoreAppCurrentTestHostSharedFrameworkPath)</IlcFrameworkPath>
+    <NoWarn>$(NoWarn);IL3050;IL3052;IL1005</NoWarn>
+
+    <!-- Forced by ILLink targets; we should fix the SDK -->
+    <SelfContained>true</SelfContained>
```
Comment (agocke) [inline eng/testing/tests.singlefile.targets:31]: Is it bad to have this always set? I presume we would eventually want to have it set in some shared targets file so people don't have to worry about it, but it seems like the property isn't exactly wrong...

**Review by agocke — APPROVED** [review]
LGTM

File: eng/testing/tests.singlefile.targets (hunk 2 above)
Comment (MichalStrehovsky) [reply eng/testing/tests.singlefile.targets:31]: It's an annoyance. I fixed the targets not to require it in https://github.com/dotnet/sdk/pull/23470.

**Pull request merged**